### DISCUSSION TITLE
How would you use "otpauth" to generate the OTP?

### DISCUSSION THREAD

**Discussion by b3ck** [zSoulweaver/kient#9] [Q&A] [answered]
Just curious how you would use "otpauth" since it needs a secret, I tried finding it on authy but I couldn't, any tips?

**Comment by zSoulweaver** [comment]
If you've already setup 2fa you'd need to find a way to export the TOTP secret from your authenticator app of choice. In your circumstance, I believe there are ways to export out of Authy but they require some specific methods, google around. Alternatively, you'd need to remove 2fa via Kick, then save your TOTP secret before re-registrating your authenticator app.

**Reply by b3ck** [reply]
okay thank you, I appreciate it, so would it be better just to deauth my kick account from Authy and use a different 2FA app that allows me to get the secret? If so do you know which app will allow me to do this off the top of your head? I'm still going to do my own research but thought I would ask =)

Thanks again,
-b3ck

**Reply by zSoulweaver** [reply, marked as answer]
> so would it be better just to deauth my kick account from Authy and use a different 2FA app that allows me to get the secret? If so do you know which app will allow me to do this off the top of your head?

Probably the easiest route, but I'm unsure if Kick enforces any sort of lockout/timeout for 2fa deregistration. You also don't need to stop using Authy, you just need to save the code that Kick provides before registering it to Authy.

**Reply by b3ck** [reply]
Yup I saw the code down under the QR and kept it just in case, registered it with 2FAS as well, I was able to get it going, now just gotta figure out the rest =)

--Thanks again for making this!
